### PR TITLE
Fix BaseAvatar wrongly retrying urls

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/riot-web/issues/13787

### CODE REVIEW

File: src/components/views/avatars/BaseAvatar.js [hunk 1]
```diff
@@ -30,54 +30,44 @@ const useImageUrl = ({url, urls, idName, name, defaultToInitialLetter}) => {
...
 
     useEffect(() => {
         // work out the full set of urls to try to load. This is formed like so:
-        // imageUrls: [ props.url, ...props.urls, default image ]
+        // imageUrls: [ props.url, ...props.urls ]
 
         let _urls = [];
         if (!SettingsStore.getValue("lowBandwidth")) {
-            _urls = urls || [];
+            _urls = memoizedUrls || [];
```
Comment (bwindels) [inline src/components/views/avatars/BaseAvatar.js:44]: Won't `memoizedUrls` be a string here? Whereas the code below assumes `_urls` is an array ...

File: src/components/views/avatars/BaseAvatar.js [hunk 2]
```diff
@@ -30,54 +30,44 @@ const useImageUrl = ({url, urls, idName, name, defaultToInitialLetter}) => {
...
     const cli = useContext(MatrixClientContext);
     const onClientSync = useCallback((syncState, prevState) => {
         // Consider the client reconnected if there is no error with syncing.
         // This means the state could be RECONNECTING, SYNCING, PREPARED or CATCHUP.
         const reconnected = syncState !== "ERROR" && prevState !== syncState;
-        if (reconnected && urlsIndex > 0 ) { // Did we fall back?
-            // Start from the highest priority URL again
-            setIndex(0);
+        if (reconnected) {
+            setIndex(i => i > 0 ? 0 : i);
```
Comment (bwindels) [inline src/components/views/avatars/BaseAvatar.js:64]: Won't this resolve to `0` either way? :thinking: Or are negative values somehow expected here?

**Comment by t3chguy** [comment]
> it's hard to figure out from the code what the problem was and how this solves it... could you elaborate a bit?

Review it commit-by-commit
https://github.com/matrix-org/matrix-react-sdk/pull/4629/commits/02ddda587b7354f90c14e082be3170117606ef6c is the fix for retrying on every sync. sync event from js-sdk has two args, currentState, prevState. Without this commit prevState was always undefined so every sync triggered the retry

File: src/components/views/avatars/BaseAvatar.js (hunk 1 above)
Comment (t3chguy) [reply src/components/views/avatars/BaseAvatar.js:44]: memoizedUrls is `() => urls` so not a string

File: src/components/views/avatars/BaseAvatar.js (hunk 2 above)
Comment (t3chguy) [reply src/components/views/avatars/BaseAvatar.js:64]: Yup, good spot!

**Pull request merged**